# A bank accepts rolls of pennies and gives 50 cents credit to a customer without counting the contents

# Assume that a roll contains 49 pennies 30% of the time, 50 pennies 60% of the time, and 51 pennies 10% of the time

## a) Find the expected value and variance for the amount that the bank loses on a typical roll

## b) Estimate the probability that the bank will lose more than 25 cents in 100 rolls

## c) Estimate the probability that the bank will lose exactly 25 cents in 100 rolls

## d) Estimate the probability that the bank will lose money in 100 rolls

## e) How many rolls does the bank need to collect to have a 99% chance of a loss?

______

# a)

# Let $X$ be the number of pennies in the roll

# Then $m(x) = \left\{\begin{matrix}0.3 & x=49\\0.6 & x=50\\0.1 & x=51 \end{matrix}\right.$

# $\implies E(X) = (0.3)(49) + (0.6)(50) + (0.1)(51) = 49.8$

# $E(X^{2}) = (0.3)(49^{2}) + (0.6)(50^{2}) + (0.1)(51^{2}) = 2480.4$

# $\implies V(X) = E(X^{2}) - (E(X))^{2} = 2480.4 - 49.8^{2} = 0.36$

# Now, we let $Y$ be the amount that the bank loses

# $\implies Y =50-X \implies E(Y) = E(50-X) = 50 - E(X) = 0.2$ and $V(Y) = V(50-X) = V(-X) = V(X) = 0.36$

_____

# b)

# $S_{100} = Y_{1} +Y_{2} + ... + Y_{100}$

# $\implies E(S_{100}) = 100 \cdot E(Y_{i}) = 20; V(S_{100}) = 100\cdot V(Y_{i}) = 36$

# $\implies \frac{25-20}{\sqrt{36}} = \frac{5}{6} = 0.8333$

# So we want to know the probability of being 0.8333 standard deviations above the mean

In [8]:
from scipy.stats import norm

In [10]:
1-norm.cdf(0.83333)

0.20232932067119946

# Therefore $P(\text{Lose more than 25 cents}) = 0.2023$

________

# c)

# If we lose exactly 25 cents, we can approximate is as losing between 24.5 cents and 25.5 cents

# $\frac{5.5}{6} = 0.916666$ and $\frac{4.5}{6} = 0.75$

In [13]:
norm.cdf(0.9166666)-norm.cdf(0.75)

0.046968665739592241

# Therefore $P(\text{Losing exactly 25 cents}) = 0.047$

______

# d)

# If the bank loses no money, $S_{100} = 0$

# $\frac{0-20}{6} = 3.3333$

In [17]:
1-norm.cdf(-3.3333)

0.99957088825464713

# So the probability of losing any amount of money is more than 99%

_______

# e)

# If we have a 99% chance of a loss, then $P(S_{n}\leq 0) = 0.01$

# $E(S_{n}) = n\cdot 0.2$ and $\sigma_{S_{n}} = \sqrt{n\cdot 0.36} = 0.6\sqrt{n}$

In [18]:
norm.ppf(0.01)

-2.3263478740408408

# That means that we need to solve for $n$ such that

# $\frac{0-0.2n}{0.6\sqrt{n}} = -2.32635 \implies \frac{\sqrt{n}}{3} = 2.32635 \implies n = 48.71 \approx 49$

# Therefore to have a 99% chance of loss, the bank needs to collect 49 rolls